src/Missings.jl

```diff
@@ -208,6 +208,91 @@ missing
 """
 passmissing(f) = PassMissing{Core.Typeof(f)}(f)
 
+struct SpreadMissings{F} <: Function
+    f::F
+end
+
+function (f::SpreadMissings{F})(xs...; kwargs...) where {F}
+    if any(x -> x isa AbstractVector{>:Missing}, xs)
+        vecs = Base.filter(x -> x isa AbstractVector, xs)
+        s = skipmissings(vecs...)
+        nonmissinginds = collect(eachindex(first(s)))
+        vecs_counter = 1
+        newargs = ntuple(length(xs)) do i
+            if xs[i] isa AbstractVector
+                t = view(xs[i], nonmissinginds)
+                vecs_counter += 1
+            else
+                t = xs[i]
+            end
+            t
+        end
+
+        res = f.f(newargs...; kwargs...)
+
+        if res isa AbstractVector
+            out = similar(res, Union{eltype(res), Missing}, length(vecs[1]))
+            fill!(out, missing)
+            out[nonmissinginds] .= res
+        else
+            out = similar(vecs[1], Union{typeof(res), Missing})
+            fill!(out, missing)
+            out[nonmissinginds] .= Ref(res)
+        end
+
+        return out
+    else
+        return f.f(xs...; kwargs...)
+    end
```

Comment:
OK. Something that occurred to me though: should we do something special if one of the non-vector arguments is `missing` (like returning `missing`)? Should we reserve this behavior for the future in case we want `spreadmissings` to be a more general version of `passmissing`?

Reply:
I don't think so. We definitely can't return `missing`, we would have to return a vector of `missing`s, which seems very restrictive.

Reply:
Not sure what "restrictive" means here. That would be similar to e.g. `broadcast(+, missing, [1, 2, 3])`.

Whether that makes sense depends on what `f` does with its inputs, but I can't find examples of functions to which one would pass one or more vectors plus a `missing` argument and would not expect the resulting vector to be full of `missing`s. So reserving the situation where one argument is `missing` could be a good idea in case we want to handle it later.